crates/solver/src/liquidity/slippage.rs

```diff
@@ -447,4 +488,41 @@ mod tests {
             assert_eq!(amount_with_slippage, expected);
         }
     }
+
+    #[test]
+    fn amm_execution_slippage() {
+        let calculator = SlippageCalculator::from_bps(100, Some(U256::exp10(18)));
+        let prices = externalprices! { native_token: WETH };
+
+        let slippage = calculator.context(&prices);
+        for (execution, expected) in [
+            (
+                AmmOrderExecution {
+                    input_max: (WETH, 1_000_000_000_000_000_000_u128.into()),
+                    output: (GNO, 10_000_000_000_000_000_000_u128.into()),
+                    internalizable: false,
+                },
+                1_010_000_000_000_000_000_u128.into(),
+            ),
+            (
+                AmmOrderExecution {
+                    input_max: (GNO, 10_000_000_000_000_000_000_000_u128.into()),
+                    output: (WETH, 1_000_000_000_000_000_000_000_u128.into()),
+                    internalizable: false,
+                },
+                10_010_000_000_000_000_000_000_u128.into(),
```

Comment:
Wouldn't 100 bps slippage of `10 GNO` mean `expected` should be `10_100_000_000_000_000_000_000_u128`?

Also do we need another test case where the absolute slippage cap kicks in or am I missing something?

Reply:
This is a case where the absolute slippage cap kicks in based on the output token (and not the input token).

Reply:
Oh, I miscounted the zeroes. I thought the tests used 1WETH and not 1000WETH. 🤦‍♂️